lib/linguist/heuristics.yml

```diff
@@ -412,6 +412,21 @@ disambiguations:
     pattern: '(?i:\bRebol\b)'
   - language: R
     pattern: '<-|^\s*#'
+- extensions: ['.re']
+  rules:
+  - language: Reason
+    pattern:
+    - '^\s*open\s+\w+\s*;$'
+    - '^\s*include\s+\w+\s*;$'
+    - '^\s*module\s+type\s'
+    - '^\s*let\s+\w+:\s+.*=.*;\s*$'
+    - '^\s*let\s+module\s\w+\s*=\s*{'
+  - language: C++
+    pattern:
+    - '^\s*#\s*include\s+<[^>]+>'
+    - '^\s*#(if|ifdef|define|pragma)\s+\w'
+    - '^\s*template\s*<'
+    - '\bstd::\w+'
```

Comment:
I'm a \bit  wary of that word boundary. 🤔 That'll also match `"std::foo"` or `'std::bar'`, which could easily be found in a program that has a string referencing something called `std::whatever`.

Unless there's a more specific and unambiguous way to match this, I'd probably omit this pattern altogether. Or at least replace `\w+` with a list of common C++ classes/functions/etc.

Reply:
Removed. `std::` can appear in a lot of contexts, so it is a bit hard to be more specific without going wild.